dvbcut: new port

mkuron opened the pull request:
#### Description

dvbcut enables cutting MPEG2 transport streams without re-encoding anything but the first and last few frames. It was originally available at http://dvbcut.sourceforge.net, but maintenance has long been taken over by the Debian package maintainer at https://github.com/bernhardu/dvbcut-deb.

The two patches included in this port have already been merged upstream, but no release has been made since.

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'printf "%s\n" "macOS `sw_vers -productVersion` `sw_vers -buildVersion` `uname -m`" "`xcodebuild -version|awk '\''NR==1{x=$0}END{print x" "$NF}'\''`"'|tee /dev/tty|pbcopy
-->
macOS 11.6 20G165 arm64
Xcode 12.5.1 12E507

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

reneeotten reviewed (changes requested):
thanks for the submission @mkuron - a few suggestions/corrections below

mkuron commented:
I don't think I've ever written a Portfile from scratch, so your comments were very helpful, @reneeotten. I have updated the pull requested and squashed accordingly. The PATH environment variable i still needed -- I suspect that dvbcut's build system isn't using the canonical way to locate the Qt5 tools. Please let me know if anything else needs to be changed.

reneeotten commented:
@mkuron thanks for the changes, this looks good! One thing I noticed is the message:
`qt.svg: Cannot open file '/opt/local/share/dvbcut/icons/dvbcut.svg', because: No such file or directory`
when starting the program. 

Indeed, looking what/where files are installed:
```
 % port contents dvbcut
Port dvbcut contains:
  /opt/local/bin/dvbcut
  /opt/local/share/applications/dvbcut.desktop
  /opt/local/share/dvbcut/dvbcut_cs.html
  /opt/local/share/dvbcut/dvbcut_de.html
  /opt/local/share/dvbcut/dvbcut_en.html
  /opt/local/share/icons/hicolor/scalable/apps/dvbcut.svg
  /opt/local/share/man/man1/dvbcut.1.gz
  /opt/local/share/mime/packages/dvbcut.xml
```

it's true that the file doesn't exist at the expected location. So you could either copy it there or try and convince the program to look at `/opt/local/share/icons/hicolor/scalable/apps/`. Other than the error message the program does start, so it might be not that crucial...

mkuron commented:
@reneeotten, thanks, fixed that too. Previously the icon was missing in the "About dvbcut" window, now it's there.

reneeotten commented:
looks good, thanks @mkuron  - in it goes!
